### PR TITLE
Implemented the UI for reroute fulfilment app(#1ttma4h)

### PR DESCRIPTION
**Pull request by disha1202**
Screenshot from 2021-11-01 15-32-29

Screenshot from 2021-11-08 17-42-25
Screenshot from 2021-11-08 17-42-35

### CODE REVIEW

File: src/views/Home.vue [hunk 1]
```diff
@@ -1,39 +1,136 @@
...
+         
+            <!-- <p>order date</p>  -->
+        
+         
+       </ion-item>
+
+       <ion-item lines="full">
+         <ion-label>
+           <ion-thumbnail>
+             <img src="https://gravatar.com/avatar/dba6bae8c566f9d4041fb9cd9ada7741?d=identicon&f=y" alt="">
```
Comment (adityasharma7) [inline src/views/Home.vue:36]: Use Image component

File: src/views/Home.vue [hunk 2]
```diff
@@ -1,39 +1,136 @@
...
     <ion-content :fullscreen="true">
-      <ion-header collapse="condense">
-        <ion-toolbar>
-          <ion-title size="large">{{ $t('Blank') }}</ion-title>
-        </ion-toolbar>
-      </ion-header>
+      <ion-text id="your-order">
+        <h1> Your Order</h1>
+      </ion-text>
+
```
Comment (adityasharma7) [inline src/views/Home.vue:9]: Remove unwanted spacing

File: src/views/Home.vue [hunk 3]
```diff
@@ -1,39 +1,136 @@
...
+         
+             <ion-badge slot="end" color="primary">status</ion-badge>
+          
+       </ion-item>
+
+       <ion-item>
+         
+           <ion-label slot="start">
+             <!-- <list-item-title> -->
+               Store pickup
```
Comment (adityasharma7) [inline src/views/Home.vue:61]: Use internationalized text throughout

File: src/views/Home.vue [hunk 4]
```diff
@@ -65,4 +162,31 @@ export default defineComponent({
 #container a {
   text-decoration: none;
 }
+#your-order{
+  position: absolute;
+width: 123px;
```
Comment (adityasharma7) [inline src/views/Home.vue:167]: Fix indentation issues

File: src/views/Home.vue [hunk 5]
```diff
@@ -1,39 +1,136 @@
...
-    </ion-header>
+    
     
     <ion-content :fullscreen="true">
-      <ion-header collapse="condense">
-        <ion-toolbar>
-          <ion-title size="large">{{ $t('Blank') }}</ion-title>
-        </ion-toolbar>
-      </ion-header>
+      <ion-text id="your-order">
```
Comment (adityasharma7) [inline src/views/Home.vue:6]: Improve id "your-order"

File: src/views/popover.vue [hunk 6]
```diff
@@ -0,0 +1,34 @@
+<template>
```
Comment (adityasharma7) [inline src/views/popover.vue:1]: Improve popover name based upon its purpose

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/Home.vue (hunk 1 above)
Comment (adityasharma7) [reply src/views/Home.vue:36]: https://github.com/hotwax/reroute-fulfilment/blob/main/src/components/Image.vue

File: src/views/ShipmentModal.vue [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+          </ion-button>
+      </ion-buttons>  
+    </ion-toolbar>   
+ </ion-header>
+ 
+ <ion-content class="ion-padding">
+     <ion-list>
+         <ion-item>
+             <ion-label position="floating">{{ $t("Street name") }}</ion-label>
+             <ion-input></ion-input>
```
Comment (adityasharma7) [inline src/views/ShipmentModal.vue:17]: Use self closing tags wherever applicable

File: src/views/ShipmentModal.vue [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+                 <ion-select>
+                     <ion-select-option>Afghanistan</ion-select-option>
+                     <ion-select-option>Albania</ion-select-option>
+                     <ion-select-option>India</ion-select-option>
+                     <ion-select-option>United States</ion-select-option>   
+                 </ion-select>
+         </ion-item>
+     </ion-list>
+ 
+     <ion-button @click="closeModal()">{{ $t("Save") }}</ion-button>
```
Comment (adityasharma7) [inline src/views/ShipmentModal.vue:51]: Improve action name based upon it's purpose

File: package.json [hunk 9]
```diff
@@ -41,7 +41,7 @@
     "@vue/cli-plugin-router": "~4.5.0",
     "@vue/cli-plugin-typescript": "~4.5.0",
     "@vue/cli-plugin-unit-jest": "~4.5.0",
-    "@vue/cli-service": "~4.5.0",
+    "@vue/cli-service": "4.5.15",
```
Comment (adityasharma7) [inline package.json:44]: Why this is needed?

File: src/views/Home.vue [hunk 10]
```diff
@@ -95,6 +99,14 @@ export default defineComponent({
       const { role } = await popover.onDidDismiss();
       console.log("onDidDismiss resolved with role", role);
     },
+
+    async openModal() {
```
Comment (adityasharma7) [inline src/views/Home.vue:103]: Improve action name based upon it's purpose

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/Home.vue [hunk 11]
```diff
@@ -1,46 +1,120 @@
...
+   IonCard,
+   IonContent,
+   IonItem,
+   IonLabel,
+   IonPage,
+   IonText,
+   IonThumbnail,
+  },
+  methods: {
+    async openPopover(ev: Event) {
```
Comment (adityasharma7) [inline src/views/Home.vue:90]: Improve action name based upon it's purpose

File: src/views/Home.vue [hunk 12]
```diff
@@ -1,46 +1,120 @@
...
+      const popover = await popoverController.create({
+        component: Popover,
+       
+        event: ev,
+        translucent: true,
+      });
+      await popover.present();
+
+      const { role } = await popover.onDidDismiss();
+      console.log("onDidDismiss resolved with role", role);
```
Comment (adityasharma7) [inline src/views/Home.vue:100]: We can remove it

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: package.json (hunk 9 above)
Comment (disha1202) [reply package.json:44]: > Why this is needed?
It updated automatically. Changed it back to version "~4.5.0"

File: src/views/Home.vue [hunk 13]
```diff
@@ -1,46 +1,104 @@
...
+   IonNote,
+   IonPage,
+   IonText,
+   IonThumbnail,
+  },
+  methods: {
+    async openPickupPopover(ev: Event) {
+      const popover = await popoverController.create({
+        component: Popover,
+        cssClass: "my-custom-class",
```
Comment (adityasharma7) [inline src/views/Home.vue:81]: Remove this

File: src/views/Home.vue [hunk 14]
```diff
@@ -1,46 +1,104 @@
...
+        event: ev,
+        translucent: true,
+      });
+      await popover.present();
+    },
+
+    async openShipmentModal() {
```
Comment (adityasharma7) [inline src/views/Home.vue:88]: Improve method name based upon it's purpose. Something like updateShipmentAddress

File: src/views/Home.vue [hunk 15]
```diff
@@ -1,46 +1,104 @@
...
+   IonContent,
+   IonItem,
+   IonLabel,
+   IonNote,
+   IonPage,
+   IonText,
+   IonThumbnail,
+  },
+  methods: {
+    async openPickupPopover(ev: Event) {
```
Comment (adityasharma7) [inline src/views/Home.vue:78]: Improve name based upon it's purpose. Something like changePickupPreference.

File: src/views/Home.vue [hunk 16]
```diff
@@ -56,13 +114,36 @@ export default defineComponent({
...
-  
   margin: 0;
 }
 
 #container a {
   text-decoration: none;
 }
-</style>
\ No newline at end of file
+#your-order {
```
Comment (adityasharma7) [inline src/views/Home.vue:124]: I think this id doesn't exist Please check.

File: src/views/ShipmentModal.vue [hunk 17]
```diff
@@ -0,0 +1,69 @@
...
+        <ion-label position="floating">{{ $t("Country") }}</ion-label>
+        <ion-select>
+          <ion-select-option>Afghanistan</ion-select-option>
+          <ion-select-option>Albania</ion-select-option>
+          <ion-select-option>India</ion-select-option>
+          <ion-select-option>United States</ion-select-option>   
+        </ion-select>
+      </ion-item>
+    </ion-list>
+    <ion-button @click="closeShipmentModal()">{{ $t("Save") }}</ion-button>
```
Comment (adityasharma7) [inline src/views/ShipmentModal.vue:44]: Improve method name closeShipmentModal based upon it's purpose. Something like save

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

**Comment by dt2patel** [comment]
Needs work on UI practices and correct usage of components. @azkyakhan please review when you have time. This is not urgent priority.

File: changelogs/unreleased/-1ttma4h.yml [hunk 18]
```diff
@@ -0,0 +1,6 @@
+---
+title: Implement static UI for the Reroute Fulfilment app
+ticket_id: "#1ttma4h"
+merge_request: https://github.com/hotwax/reroute-fulfilment/pull/6/commits/0a15531748e45880cd698c2d2b59821171321af7
```
Comment (azkyakhan) [inline changelogs/unreleased/-1ttma4h.yml:4]: Please use MR no. here instead of link

File: src/locales/en.json [hunk 19]
```diff
@@ -1,11 +1,27 @@
 {
   "Blank": "Blank",
```
Comment (azkyakhan) [inline src/locales/en.json:2]: Arrange all the texts here in alphabetical order

File: src/views/ChangePickupPreferencePopover.vue [hunk 20]
```diff
@@ -0,0 +1,34 @@
...
+      <ion-item lines="none">
+        <ion-icon slot="start" :icon="mailOutline" />
+        <ion-label>{{ $t("Get it shipped") }}</ion-label>
+      </ion-item>     
+    </ion-list>
+  </ion-content>
+</template>
+
+<script>
+import { IonContent, IonList, IonItem, IonIcon, IonLabel } from '@ionic/vue';
```
Comment (azkyakhan) [inline src/views/ChangePickupPreferencePopover.vue:17]: Arrange component here in alphabetical order

File: src/views/Home.vue [hunk 21]
```diff
@@ -56,13 +111,29 @@ export default defineComponent({
...
-  
   color: #8c8c8c;
-  
   margin: 0;
 }
 
 #container a {
   text-decoration: none;
 }
-</style>
```
Comment (azkyakhan) [inline src/views/Home.vue:68]: Please remove unwanted styling

File: src/views/ChangePickupPreferencePopover.vue [hunk 22]
```diff
@@ -0,0 +1,34 @@
...
+      </ion-item>     
+    </ion-list>
+  </ion-content>
+</template>
+
+<script>
+import { IonContent, IonIcon, IonLabel, IonList, IonItem } from '@ionic/vue';
+import { defineComponent } from 'vue';
+import { 
+ storefrontOutline,
```
Comment (ymaheshwari1) [inline src/views/ChangePickupPreferencePopover.vue:20]: Improve indentation

File: src/views/Home.vue [hunk 23]
```diff
@@ -1,68 +1,100 @@
...
+        <ion-item lines="none">
+          <ion-label>
+            <h2>Customer Name</h2>
+            <p>Order ID</p>
+          </ion-label>
+          <ion-note slot="end">order date</ion-note>        
+        </ion-item>
+        <ion-item lines="full">
+          <ion-thumbnail slot="start">
+            <Image src="https://gravatar.com/avatar/dba6bae8c566f9d4041fb9cd9ada7741?d=identicon&f=y"></Image>
```
Comment (ymaheshwari1) [inline src/views/Home.vue:17]: Use self-closing tag.

File: src/views/ChangePickupPreferencePopover.vue [hunk 24]
```diff
@@ -0,0 +1,34 @@
+<template>
+  <ion-content class="ion-padding">
```
Comment (dt2patel) [inline src/views/ChangePickupPreferencePopover.vue:2]: remove ion padding @disha1202

File: src/views/Home.vue [hunk 25]
```diff
@@ -1,68 +1,100 @@
...
-        <ion-toolbar>
-          <ion-title size="large">{{ $t('Blank') }}</ion-title>
-        </ion-toolbar>
-      </ion-header>
-    
-      <div id="container">
-        <strong>{{ $t('Ready to create an app?') }}</strong>
-        <p>{{ $t('Start with Ionic') }} <a target="_blank" rel="noopener noreferrer" href="https://ionicframework.com/docs/components">{{ $t('UI Components') }}</a></p>
-      </div>
+      <ion-text id="title">
```
Comment (dt2patel) [inline src/views/Home.vue:4]: why did you need to use ion-text here?

File: src/views/Home.vue [hunk 26]
```diff
@@ -1,68 +1,100 @@
...
+          <ion-thumbnail slot="start">
+            <Image src="https://gravatar.com/avatar/dba6bae8c566f9d4041fb9cd9ada7741?d=identicon&f=y" />
+          </ion-thumbnail>
+          <ion-label slot="start">
+            <p>BRAND</p>
+            <h2>Virtual Name</h2>
+            <p>{{ $t("Color") }}: color</p>
+            <p>{{ $t("Size") }}: size</p>
+          </ion-label>
+          <ion-badge slot="end" color="primary">status</ion-badge>
```
Comment (dt2patel) [inline src/views/Home.vue:25]: do you need to explicitly state primary color?

File: src/views/Home.vue [hunk 27]
```diff
@@ -1,68 +1,100 @@
...
+        })
+      return modal.present();
+    },
+  },
 });
 </script>
 
 <style scoped>
-#container {
+#title h1 {
```
Comment (dt2patel) [inline src/views/Home.vue:97]: do you need a special ID here?

**Review by dt2patel — CHANGES_REQUESTED** [review]

**Comment by disha1202** [comment]
@dt2patel Sir I have done all the requested changes.

**Review by adityasharma7 — APPROVED** [review]
